extensions/policy-monitor/store/sql/policy-monitor-store-sql/src/main/java/org/eclipse/edc/connector/policy/monitor/store/sql/SqlPolicyMonitorStore.java

```diff
@@ -0,0 +1,180 @@
+        super(dataSourceRegistry, dataSourceName, transactionContext, objectMapper, queryExecutor);
+        this.statements = statements;
+        this.clock = clock;
+        this.leaseHolderName = leaseHolderName;
+        leaseContext = SqlLeaseContextBuilder.with(transactionContext, leaseHolderName, statements, clock, queryExecutor);
+    }
+
+    @Override
+    public @Nullable PolicyMonitorEntry findById(String id) {
+        return transactionContext.execute(() -> {
+            try (var connection = getConnection()) {
+                return findByIdInternal(connection, id);
+            } catch (SQLException e) {
+                throw new EdcPersistenceException(e);
+            }
+        });
+    }
+
+    @Override
+    public @NotNull List<PolicyMonitorEntry> nextNotLeased(int max, Criterion... criteria) {
+        return transactionContext.execute(() -> {
+            var filter = Arrays.stream(criteria).collect(toList());
+            var querySpec = QuerySpec.Builder.newInstance().filter(filter).limit(max).build();
+            var statement = statements.createQuery(querySpec)
+                    .addWhereClause(statements.getNotLeasedFilter(), clock.millis());
+
+            try (
+                    var connection = getConnection();
+                    var stream = queryExecutor.query(connection, true, this::mapEntry, statement.getQueryAsString(), statement.getParameters())
+            ) {
+                var entries = stream.collect(Collectors.toList());
+                entries.forEach(entry -> leaseContext.withConnection(connection).acquireLease(entry.getId()));
+                return entries;
+            } catch (SQLException e) {
+                throw new EdcPersistenceException(e);
+            }
+        });
+    }
+
+    @Override
+    public StoreResult<PolicyMonitorEntry> findByIdAndLease(String id) {
+        return transactionContext.execute(() -> {
+            try (var connection = getConnection()) {
+                var entity = findByIdInternal(connection, id);
+                if (entity == null) {
+                    return StoreResult.notFound(format("DataFlow %s not found", id));
+                }
+
+                leaseContext.withConnection(connection).acquireLease(entity.getId());
+                return StoreResult.success(entity);
+            } catch (IllegalStateException e) {
+                return StoreResult.alreadyLeased(format("DataFlow %s is already leased", id));
+            } catch (SQLException e) {
+                throw new EdcPersistenceException(e);
+            }
+        });
+    }
+
+    @Override
+    public void save(PolicyMonitorEntry entity) {
+        transactionContext.execute(() -> {
+            try (var connection = getConnection()) {
+                var existing = findByIdInternal(connection, entity.getId());
+                if (existing != null) {
+                    leaseContext.by(leaseHolderName).withConnection(connection).breakLease(entity.getId());
```

Comment:
minor 

`leaseContext.by(leaseHolderName)` could be a noop since the `leaseHolder` should not change right?

Reply:
🤣